### PR TITLE
Bootstrap widgets with local JS data

### PR DESCRIPTION
**Pull request by joshkadis**
And allow updates by triggering an action on the widget element

### CODE REVIEW

File: app/actions/index.js [hunk 1]
```diff
@@ -95,3 +98,32 @@ export function bootstrapWidgetData(widgetId, fetchUrl, headersAttr = null) {
       .then(handleJson);
   };
 }
+
+export function listenerWidgetData(widgetEl, store) {
```
Comment (joshkadis) [inline app/actions/index.js]: @jackmarketon Cool to pass the store to this function?

**Comment by danbowles** [comment]
🐴

File: app/actions/index.js (hunk 1 above)
Comment (joshkadis) [inline app/actions/index.js]: now just passing `dispatch` instead of the whole `store`

**Comment by joshkadis** [comment]
Could still use a review of those last few commits, thanks.

File: app/widget.js [hunk 2]
```diff
@@ -4,7 +4,8 @@ import ReactDOM from 'react-dom';
 import { Provider } from 'react-redux';
 import { createStore, applyMiddleware, compose } from 'redux';
 import thunk from 'redux-thunk';
-import { bootstrapWidgetData } from './actions';
+import actionTrigger, { ajaxWidgetData, listenerWidgetData } from './actions';
```
Comment (nosro) [inline app/widget.js]: it seemed a little odd to have the mixed exports from actions, with `actionTrigger` as default, then the rest.

File: README.md [hunk 3]
```diff
@@ -68,3 +70,88 @@ Note that our transformer changed the keys to `label` and `value`, and the value
...
+
+##### Synchronous data
+
+Use the global `_SimplechartWidgetData` with the widget's `id` as a key. In this case, `_SimplechartWidgetData.examplewidget` should include all of the data needed to render the chart, including options and metadata.
+
+You can only use this method once per widget, _before_ loading `widget.js`.
+
+##### Asynchronous data
+
+At any time, you can trigger the event `widgetData` on the widget's containing element. If `evt.detail` contains `data`, `options`, _and_ `metadata`, the entire chart will be rebuilt.
```
Comment (nosro) [inline README.md]: Just curious why not Simplechart in the event name too, like the global data structure above? Too long?

**Pull request merged**